plugins/services/src/js/containers/services/ServiceTreeView.js

```diff
@@ -54,6 +57,52 @@ class ServiceTreeView extends React.Component {
     );
   }
 
+  getNoLimitInfobox() {
+    const { serviceTree } = this.props;
+
+    // TODO: remove feature flag
+    const { quota } = ConfigStore.get("config").uiConfiguration.features || {};
+    const { rolesCount, groupRolesCount } = serviceTree.getRolesLength();
+
+    if (
+      !quota ||
+      !serviceTree ||
+      serviceTree.id === "/" ||
+      !serviceTree.getEnforceRole() ||
+      rolesCount === groupRolesCount
+    ) {
+      return null;
+    }
+
+    return (
+      <InfoBoxInline
+        className="quota-info"
+        appearance="default"
+        message={
+          <React.Fragment>
+            <Icon
+              shape={SystemIcons.CircleInformation}
+              size={iconSizeXs}
+              color="currentColor"
+            />
+            {serviceTree.list.length === 1 ? (
+              <Trans render="span">
+                1 of 1 service does not have limit. Please upgrade to be
+                included in quota.
+              </Trans>
+            ) : (
+              <Trans render="span">
+                {rolesCount - groupRolesCount} of {serviceTree.list.length}{" "}
```

Comment:
maybe `serviceTree.getRolesLength` can be renamed and directly return this number? i imagine something like a `servicesWithoutLimitCount`.

Reply:
We need the two numbers in other cases, otherwise we could just return the string (as it was in my original version).